src/checkpoint/checkpoint_api.impl.h

```diff
@@ -91,6 +91,12 @@ std::size_t getSize(T& target) {
   return dispatch::Standard::size<T, Sizer>(target);
 }
 
+template <typename T>
+std::size_t getMemoryFootprint(T& target) {
+  // FIXME: use std::max to make sure that it's not smaller than sizeof(T)
```

Comment:
at this stage (of implementation) `getMemoryFootprint` sometimes returns smaller value than a simple `sizeof`
(this can be valid calculation because of padding / layout)

note to self: change to `std::max(footprint, sizeof(T))` before merging